server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -317,6 +329,31 @@ public String executor() {
         }
     }
 
+    TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) throws Exception {
+        // Read enum
+        switch (extensionRequest.getRequestType()) {
+            case REQUEST_EXTENSION_CLUSTER_STATE:
+                ClusterStateResponse clusterStateResponse = new ClusterStateResponse(
+                    clusterService.getClusterName(),
+                    clusterService.state(),
+                    false
+                );
+                return clusterStateResponse;
+            case REQUEST_EXTENSION_LOCAL_NODE:
+                LocalNodeResponse localNodeResponse = new LocalNodeResponse(clusterService);
+                return localNodeResponse;
+            case REQUEST_EXTENSION_CLUSTER_SETTINGS:
+                ClusterSettingsResponse clusterSettingsResponse = new ClusterSettingsResponse(clusterService);
+                return clusterSettingsResponse;
+            case REQUEST_EXTENSION_ACTION_LISTENER_ON_FAILURE:
+                listener.onFailure(new Exception(extensionRequest.getFailureException()));
```

Comment:
Instead of calling the listener here. We should create a separate response to handle this.

Reply:
(If it stays) More narrowly type this exception.  (If it moves, more narrowly type the new exception in the new code.)

Reply:
Do you have a specific type in mind?  The actual type of the exception exists in the exception message itself since there is no way to pass over exceptions through transport service.

Reply:
> Any particular reason why? The response would just be calling the exact same line.

Because we want to follow the same code style we have for other cases as well.
The way we could have called `localNode` [here](https://github.com/opensearch-project/OpenSearch/blob/feature/extensions/server/src/main/java/org/opensearch/cluster/LocalNodeResponse.java#L29) itself but we didn't .